pages/news.tsx

```diff
@@ -0,0 +1,82 @@
+import { FC, useEffect, useState } from 'react'
+import 'twin.macro'
+import Head from 'next/head'
+import { GetServerSideProps } from 'next'
+import axios from 'axios'
+import { POST_LIMIT } from '../constants/news'
+import { useRouter } from 'next/router'
+import { Post } from '../types/post'
+import PostLayout from '../components/postLayout'
+import useSWR from 'swr'
+import axiosFetcher from '../lib/axiosFetcher'
+import Loader from '../components/loader'
+
+export const getServerSideProps: GetServerSideProps = async ({
+  query,
+  params,
+}) => {
+  let posts
+
+  try {
+    const response = await axios.get('/trending/feed', {
+      params: { limit: POST_LIMIT, page: params?.page || 1 },
+    })
+
+    if (response.status === 200) {
+      posts = await response.data
+    } else {
+      posts = []
+    }
+  } catch (e) {
+    posts = []
+  }
```

Comment:
Можна трохи спростити код, якщо `post` ініціалізувати як пустий масив

```suggestion
  let posts = []

  try {
    const response = await axios.get('/trending/feed', {
      params: { limit: POST_LIMIT, page: params?.page || 1 },
    })

    if (response.status === 200) {
      posts = await response.data
    }
  } catch (e) {
    // log or show error
  }
```

Reply:
Тут согласен, что нужно упростить, но хардкодить 200 это не клёво, потому что тебе может прийти и 204, а ты это код не обрабатываешь:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses